lib/pp.rb

```diff
@@ -149,6 +150,10 @@ def pop_inspect_key(id)
     # Object#pretty_print_cycle is used when +obj+ is already
     # printed, a.k.a the object reference chain has a cycle.
     def pp(obj)
+      # If obj is a Delegator then use the object being delegated to for cycle
+      # detection
+      obj = obj.__getobj__ if obj.is_a? Delegator
```

Comment:
`defined?(::Delegator) and obj.is_a?(::Delegator)` would be better than requiring 'delegate' always, I think.

Reply:
Shall I make this change?

Reply:
I see there are also now conflicts on this PR, can have a look if that'd be useful.